src/plugins/expression_shape/public/components/progress/progress_component.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { CSSProperties, RefCallback, useCallback, useEffect, useRef, useState } from 'react';
+import { useResizeObserver } from '@elastic/eui';
+import { IInterpreterRenderHandlers } from '../../../../expressions';
+import { NodeDimensions, ProgressRendererConfig } from '../../../common/types';
+import { ShapeRef, SvgConfig, SvgTextAttributes } from '../reusable/types';
+import { getShapeContentElement } from '../reusable/shape_factory';
+import { getId, withSuspense } from '../../../../presentation_util/public';
+import { getTextAttributes, getViewBox } from './utils';
+import { getDefaultShapeData } from '../reusable';
+import { LazyProgressDrawer } from '../..';
+
+const ProgressDrawer = withSuspense(LazyProgressDrawer);
+
+interface ProgressComponentProps extends ProgressRendererConfig {
+  onLoaded: IInterpreterRenderHandlers['done'];
+  parentNode: HTMLElement;
+}
+
+function ProgressComponent({
+  onLoaded,
+  parentNode,
+  shape: shapeType,
+  value,
+  max,
+  valueColor,
+  barColor,
+  valueWeight,
+  barWeight,
+  label,
+  font,
+}: ProgressComponentProps) {
+  const parentNodeDimensions = useResizeObserver(parentNode);
+  const [dimensions, setDimensions] = useState<NodeDimensions>({
+    width: parentNode.offsetWidth,
+    height: parentNode.offsetHeight,
+  });
+  const [shapeData, setShapeData] = useState<SvgConfig>(getDefaultShapeData());
+  const shapeRef = useCallback<RefCallback<ShapeRef>>((node) => {
+    if (node !== null) {
+      setShapeData(node.getData());
+    }
+  }, []);
+
+  const [totalLength, setTotalLength] = useState<number>(0);
+
+  useEffect(() => {
+    setDimensions({
+      width: parentNode.offsetWidth,
+      height: parentNode.offsetHeight,
+    });
+    onLoaded();
+  }, [onLoaded, parentNode, parentNodeDimensions]);
+
+  const progressRef = useRef<
+    SVGCircleElement & SVGPathElement & SVGPolygonElement & SVGRectElement
+  >(null);
+  const textRef = useRef<SVGTextElement>(null);
+
+  useEffect(() => {
+    setTotalLength(progressRef.current ? progressRef.current.getTotalLength() : 0);
+  }, [shapeType, shapeData, progressRef]);
+
+  const BarProgress = shapeData.shapeType ? getShapeContentElement(shapeData.shapeType) : null;
+
+  const shapeContentAttributes = {
+    className: 'canvasProgress__background',
```

Comment:
We need to get rid of all these `canvas_` prefixed class names since this is not strictly canvas anymore.  Are they even used for anything at this point? Can you double check on that?

Reply:
Removed this `canvas*` classes. According to the code, it is not necessary, but, possibly, some tests were using it. After passing the CI we'll see)))